pkg/validate/raw/hydrate/declared_field_hydrator.go

```diff
@@ -134,6 +134,13 @@ func setDefaultProtocol(u *unstructured.Unstructured) status.MultiError {
 	case kinds.CronJob().GroupKind():
 		errs = setDefaultProtocolInNestedPodSpec(u.Object, "spec", "jobTemplate", "spec", "template", "spec")
 	case kinds.Service().GroupKind():
+		serviceType, found, err := unstructured.NestedString(u.Object, "spec", "type")
+		if err != nil {
+			errs = append(errs, err)
+		}
+		if found && serviceType == "ExternalName" {
+			break
+		}
```

Comment:
Actually, Kubernetes doesn't say `Ports` is a required fields for `Service`: https://github.com/kubernetes/kubernetes/blob/fb57080e069fe2efe11734a4570a92873e673c35/pkg/apis/core/v1/defaults.go#L114-L122 

Therefore, please fix `func setDefaultProtocolInNestedPorts` instead. The part that assumes `ports` must be set in `Service` isn't correct.

Reply:
So instead we don't need to check ports here for services? Can make the change

Reply:
Yes. The logic should be similar to the LOC I linked. If `Ports` isn't specified it will just return.

Reply:
Okie removed the port check for Service, updated the commit message and the TODO comment as well.

Reply:
Don't forget to update PR description as well